src/fiber/types.rs

```diff
@@ -2592,6 +2607,42 @@ pub enum BroadcastMessageID {
     NodeAnnouncement(Pubkey),
 }
 
+// We need to implement Ord for BroadcastMessageID to make sure that a ChannelUpdate message is always ordered after ChannelAnnouncement,
+// so that we can use it as the sorting key in fn prune_messages_to_be_saved to simplify the logic.
+impl Ord for BroadcastMessageID {
+    fn cmp(&self, other: &Self) -> Ordering {
+        match (self, other) {
+            (
+                BroadcastMessageID::ChannelAnnouncement(outpoint1),
+                BroadcastMessageID::ChannelAnnouncement(outpoint2),
+            ) => outpoint1.cmp(outpoint2),
+            (
+                BroadcastMessageID::ChannelUpdate(outpoint1),
+                BroadcastMessageID::ChannelUpdate(outpoint2),
+            ) => outpoint1.cmp(outpoint2),
+            (
+                BroadcastMessageID::NodeAnnouncement(pubkey1),
+                BroadcastMessageID::NodeAnnouncement(pubkey2),
+            ) => pubkey1.cmp(pubkey2),
+            (BroadcastMessageID::ChannelUpdate(_), _) => Ordering::Less,
+            (BroadcastMessageID::NodeAnnouncement(_), _) => Ordering::Greater,
```

Comment:
Did you intend to order messages by making `NodeAnnouncement` in the first place, `ChannelAnnouncement` the second place and `ChannelUpdate` the last place (this way the sorted messages in the list `messages_to_be_saved` can be processed sequentially)? If that is the case, I think the correct Ordering should be https://github.com/contrun/fiber/commit/ea5c1fede9bffd8d3f3fa49d2570aadb971fcb6f. This problem is not discovered in the test because another bug above always excluding `ChannelUpdate` from being included in the complete messages when there is a corresponding `ChannelAnnouncement` in the `messages_to_be_saved`.